src/jdk.sctp/unix/classes/sun/nio/ch/sctp/SctpNet.java

```diff
@@ -91,41 +89,14 @@ static Set<SocketAddress> getLocalAddresses(int fd)
         SocketAddress[] saa = getLocalAddresses0(fd);
 
         if (saa != null) {
-            set = getRevealedLocalAddressSet(saa);
+            set = new HashSet<>(saa.length);
+            for (SocketAddress sa : saa)
+                set.add(sa);
```

Comment:
It would be nice if this could be changed to `Set.of(saa)` but SctpChannel::getAllLocalAddresses doesn't specify anything about the set, so in theory there may be code that modifies the set, in which case what you have is okay.

Reply:
> [...] so in theory there may be code that modifies the set [...]

That is precisely why I did not use `Set.of(saa)`.